src/wallet/api.rs

```diff
@@ -136,6 +136,12 @@ pub enum UTXOSpendInfo {
     },
 }
 
+// Custom type to handle complex return values.
+type SwapCoinsInfo<'a> = (
+    Vec<(&'a IncomingSwapCoin, ListUnspentResultEntry)>,
+    Vec<(&'a OutgoingSwapCoin, ListUnspentResultEntry)>,
+);
+
```

Comment:
Just to make sure, is using lifetimes here intentional?

Reply:
Yeah. Could have omitted, but chose not to as it's not used much in the lib and would require cloning large structs. We might just remove this later on altogether.